tests/python/test_bitmasked.py

```diff
@@ -0,0 +1,30 @@
+import taichi as ti
+
+
+@ti.archs_excluding(ti.opengl)
+def test_basic():
+    x = ti.var(ti.i32)
+    c = ti.var(ti.i32)
+    s = ti.var(ti.i32)
+
+    bm = ti.root.dense(ti.ij, (3, 6)).bitmasked().dense(ti.i, 5).bitmasked()
+    bm.place(x)
+    ti.root.place(c, s)
+
+    @ti.kernel
+    def run():
+        x[5, 1] = 2
+        x[9, 4] = 20
+        x[0, 3] = 20
+
+    @ti.kernel
+    def sum():
+        for i, j in x:
+            ti.atomic_add(c[None], ti.is_active(bm, [i, j]))
```

Comment:
So, the difference `bitmasked` introduced is, only those assigned slots are marked as `is_active`?

Reply:
If I'm thinking right, this can be used for performance optimization, like Game of Life - only the non-static cells are calculated?

Reply:
Exactly. For sparse data structures, `struct_for` will only loop over the active indices (it could still iterate over more slots due to POT paddings, but not all of them).

Reply:
> If I'm thinking right, this can be used for performance optimization, like Game of Life - only the non-static cells are calculated?

Great idea! If you are interested, having a game of life demo with sparse grids would be super cool!!